src/ui/components/UserRating/index.js

```diff
@@ -32,11 +32,11 @@ export const UserRatingBase = (props: Props) => {
   return (
     <Rating
       className={className}
-      isOwner={isOwner}
       onSelectRating={onSelectRating}
       rating={(review && review.rating) || 0}
-      readOnly={readOnly}
+      readOnly={readOnly || false}
```

Comment:
Would it make sense to have a default value for the `readOnly` prop instead?

Reply:
We could but it won't make this Flow error go away:

```
Cannot create Rating element because undefined [1] is incompatible with boolean [2] in property readOnly.

     src/ui/components/UserRating/index.js
 [1] 15│   readOnly?: boolean,
       :
     39│     } = this.props;
     40│
     41│     return (
     42│       <Rating
     43│         className={className}
     44│         onSelectRating={onSelectRating}
     45│         rating={(review && review.rating) || 0}
     46│         readOnly={readOnly}
     47│         styleSize={styleSize}
     48│         yellowStars={isOwner}
     49│       />
     50│     );
     51│   }
     52│ }

     src/ui/components/Rating/index.js
 [2] 21│   readOnly: boolean,
```

The reason is that while Flow does understand `defaultProps` in normal cases, it doesn't know about `defaultProps` in our [type hint](https://github.com/mozilla/addons-frontend/blob/def32ce267fdd4c6aadaefd70237f381573a6456/src/ui/components/UserRating/index.js#L51) (an abnormal case) so we are forced to make `readOnly` an optional property.

I looked into making the type hint also know about `defaultProps` but I couldn't get it to work. I don't think we have a granular enough [type helper](https://flow.org/en/docs/react/types/) for it. [`React.ElementConfig`](https://flow.org/en/docs/react/types/#toc-react-elementconfig) is what we want but it doesn't let us swap `InternalProps` for `Props`.